src/value/mod.rs

```diff
@@ -483,3 +484,23 @@ pub trait Struct {
         field_name: T,
     ) -> Box<dyn Iterator<Item = &'a Self::Element> + 'a>;
 }
+
+pub trait Builder {
+    type Element: Element + ?Sized;
+    type Sequence: Sequence<Element = Self::Element> + ?Sized;
+
+    /// Build a `null` from IonType using Builder
+    fn new_null(e_type: IonType) -> Self::Element;
+
+    /// Build a `clob` using Builder
+    fn new_clob(bytes: &'static [u8]) -> Self::Element;
+
+    /// Build a `blob` using Builder
+    fn new_blob(bytes: &'static [u8]) -> Self::Element;
+
+    /// Build a `list` from Sequence using Builder
+    fn new_list(seq: Self::Sequence) -> Self::Element;
+
+    /// Build a `sexp` from Sequence using Builder
+    fn new_sexp(seq: Self::Sequence) -> Self::Element;
```

Comment:
Should these instead take `S where S: Into<Self::Sequence>`? (If so, I'm happy with tracking this in another issue.)

Reply:
We are punting on this according to an offline discussion. We already have `Sequence` as the concrete associated type here.